code/modules/surgery/organs/tongue.dm

```diff
@@ -75,8 +76,11 @@
 	say_mod = "hisses"
 	taste_sensitivity = 10 // combined nose + tongue, extra sensitive
 	modifies_speech = TRUE
+	languages_native = list(/datum/language/draconic)
 
 /obj/item/organ/tongue/lizard/handle_speech(datum/source, list/speech_args)
```

Comment:
speech_args is the list of arguments to the say() proc, to be exact the say() proc from living_say.dm https://github.com/tgstation/tgstation/blob/master/code/modules/mob/living/living_say.dm#L95

handle_speech is triggered by the COMSIG_MOB_SAY signal/event, sent here https://github.com/tgstation/tgstation/blob/master/code/modules/mob/living/living_say.dm#L197

If you notice, the language arg is set well before that signal, at https://github.com/tgstation/tgstation/blob/master/code/modules/mob/living/living_say.dm#L160. It's set in such a way that respects both speech args, and the mob's default language.

You could use the speech_args list to pull the mob's actual selected language, and use that. If you look at other things that access this list, they do so used some defines.
https://github.com/tgstation/tgstation/blob/master/code/__DEFINES/dcs/signals.dm#L561

Uncomment the one you want to use, and use it as a define. That should work fine.

Reply:
I've actially tried. My commit (actually second one, was just force-pushed) has name of "second edition", because the first one was literally a shitcode around signals. Working, but shitcode. I don't understand signals and args, so don't mess with them (but tried at first edition).

Alright, I'll try

Reply:
If you've got questions, feel free to ask. Bug me on discord if it needs more back and forth stuff. I go by some variety of potato in coderbus and on tgstation

Reply:
Done

Reply:
Done